### PR TITLE
Add-deny-equivalent-of-assertCollection-hasSameElements

### PR DESCRIPTION
**Pull request by jecisc**
Add #denyCollection:hasSameElements to assertions

### CODE REVIEW

File: src/SUnit-Core-Traits/TAssertable.trait.st [hunk 1]
```diff
@@ -194,6 +194,15 @@ TAssertable >> denyCollection: actual equals: expected [
 		description: [ self unexpectedEqualityStringBetween: actual and: expected ]
 ]
 
+{ #category : #asserting }
+TAssertable >> denyCollection: actual hasSameElements: expected [
+	"Deny that a collection contains the same elements as the given collection. Order is not checked, only the presence/absence of elements."
+
+	self
+		deny: ((actual difference: expected) isEmpty or: [ (expected difference: actual) isEmpty ])
+		description: (String streamContents: [ :stream | stream nextPutAll: 'Given collections match!' ])
```
Comment (gcotelli) [inline src/SUnit-Core-Traits/TAssertable.trait.st]: Why `String streamContents:` and not directly: `'Given collections match!'` ?

File: src/SUnit-Core/TestAsserter.class.st [hunk 2]
```diff
@@ -376,6 +376,15 @@ TestAsserter >> denyCollection: actualCollection equals: expectedCollection [
 		description: [ self unexpectedEqualityStringBetween: actualCollection and: expectedCollection ]
 ]
 
+{ #category : #asserting }
+TestAsserter >> denyCollection: actual hasSameElements: expected [
+	"Deny that a collection contains the same elements as the given collection. Order is not checked, only the presence/absence of elements."
+
+	self
+		deny: ((actual difference: expected) isEmpty or: [ (expected difference: actual) isEmpty ])
+		description: (String streamContents: [ :stream | stream nextPutAll: 'Given collections match!' ])
```
Comment (gcotelli) [inline src/SUnit-Core/TestAsserter.class.st]: Why String streamContents: and not directly: 'Given collections match!' ?

**Review by gcotelli — CHANGES_REQUESTED** [review]
This need some test cases to verify proper behavior

File: src/SUnit-Core-Traits/TAssertable.trait.st (hunk 1 above)
Comment (gcotelli) [reply src/SUnit-Core-Traits/TAssertable.trait.st]: I think the implementation is not right or I'm misinterpreting the intention.
If you do: `self denyCollection: #(1 2) hasSameElements: #(1 2 4)` the test fails

**Comment by jecisc** [comment]
@gcotelli Thanks! It's now fixed.

**Review by gcotelli — APPROVED** [review]
Looks good now

**Review by MarcusDenker — APPROVED** [review]

**Pull request merged**